src/citrine/resources/ara_definition.py

```diff
@@ -124,6 +126,74 @@ def add_columns(self, *,
             columns=copy(self.columns) + columns
         )
 
+    def add_all_ingredients(self, *,
+                            process_template: LinkByUID,
+                            project,
+                            quantity_dimension: IngredientQuantityDimension,
+                            scope: str = 'id'
+                            ):
```

Comment:
Are we saying that we only want to support complete lists of ingredients in the scenario that the Process Template has `allowed_names` set?

Reply:
That's my understanding.

The alternative would be to find the process based on this template _for each row_ of the table and aggregate the names of all the ingredients that feed into those processes. But that presents two problems:
1. I don't think the client can get those processes (though we could add an endpoint to do so)
2. What if the user then adds another row that has different ingredient names for the associated process?

We could also find every process in the project that references this process template, but then we could wind up with an obscenely long list of ingredients (and again, I don't think the client can perform that operation).

Reply:
For the convenience method approach, you are right.  I was pushing back on that assumption.

Reply:
The way that I did this for calcite was basically this:
"The alternative would be to find the process based on this template for each row of the table and aggregate the names of all the ingredients that feed into those processes."

But their complete ingredient set for the lighthouse project is a relatively limited set of ingredients. <20